### PR TITLE
feat(Image): Replaced ImageCache(UWPToolkit) by ImagePipeline(Fresco).

### PR DESCRIPTION
**Pull request by phongcao**
New features and bug fixes:
- Enable memory cache for images.
- Use memory pool, score-based eviction mechanism (size + age) instead of instance count + fixed duration.
- Postprocessor support (needed for tint/background color implementation in Win10 pre-anniversary build).
- Fix bug #1259

### CODE REVIEW

**Comment by rozele** [comment]
@phongcao sorry, the CI failure is not your problem. It's related to #1313. I've pushed #1314 to get around this.

**Comment by rozele** [comment]
Hey @phongcao, you should join the Microsoft GitHub org, you shouldn't need to sign the CLA.

---
In reply to: [323537025](https://github.com/Microsoft/react-native-windows/pull/1312#issuecomment-323537025) [](ancestors = 323537025)

**Comment by phongcao** [comment]
@rozele Sorry, I've just found a bug in ReleaseBundle and fixed it. Please review again.

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs [hunk 1]
```diff
@@ -278,54 +267,31 @@ private void OnImageStatusUpdate(Border view, ImageLoadStatus status, ImageMetad
...
+                var uri = new Uri(source);
 
-                imageBrush.ImageSource = image;
-            }
-            else if (BitmapImageHelpers.IsHttpUri(source))
-            {
-                OnImageStatusUpdate(view, ImageLoadStatus.OnLoadStart, default(ImageMetadata));
-                try
+                // Base64 or local images
+                if (source.StartsWith("data:") || source.StartsWith("ms-appx:"))
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: Good catch. When people use CodePush, the assets may also come in using the ms-appdata: prefix. Not sure what would happen in those cases.

**Comment by rozele** [comment]
Please rebase to get the latest from master so the unit tests will pass.

---
In reply to: [323539918](https://github.com/Microsoft/react-native-windows/pull/1312#issuecomment-323539918) [](ancestors = 323539918)

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs (hunk 1 above)
Comment (phongcao) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: Yeah, it's supported in Fresco but I'll need to make some small modification that works for most cases. Thanks for telling me.

**Comment by phongcao** [comment]
@rozele Seems to be working now.

**Pull request merged**